src/cdk/stepper/stepper.ts

```diff
@@ -112,6 +126,22 @@ export class CdkStep implements OnChanges {
   }
   private _optional = false;
 
+  /** State of the step. */
+  @Input()
+  get state(): StepState | string | null { return this._state; }
```

Comment:
I am a bit confused on the typings for `state`, wouldn't it be best to capture all of the appropriate states in `StepState` instead of allowing arbitrary `string`s?   Additionally, should `null` be possible, is one of the `StepState`s a reasonable default?

Reply:
Yes, makes sense! So, I've added `string` as part of the `StepState` type. And correct, it should probably never be `null`; therefore, I've added `STEP_STATE.NUMBER` as the default. Thanks!